src/Contacts/TSThread.m

```diff
@@ -248,9 +248,11 @@ - (NSString *)lastMessageLabel {
 - (void)updateWithLastMessage:(TSInteraction *)lastMessage transaction:(YapDatabaseReadWriteTransaction *)transaction {
     NSDate *lastMessageDate = lastMessage.date;
 
-    if ([lastMessage isKindOfClass:[TSIncomingMessage class]]) {
```

Comment:
> Huh, well seems to me that we should be using these dates for the conversation ordering as well.

"these dates" **are** used for sorting, but that's not in this method. The sorting is in the YapDatabaseView.